### ISSUE TITLE
[java] False positive from JUnitTestsShouldIncludeAssert with SoftAssertions

### ISSUE THREAD

**Issue by ctoomey** [pmd/pmd#4113]
**Affects PMD Version:** 

6.48.0

**Rule:**

[JUnitTestsShouldIncludeAssert](https://pmd.sourceforge.io/pmd-6.49.0/pmd_rules_java_bestpractices.html#junittestsshouldincludeassert)

**Description:**

Rule has false positives for some ways of using AssertJ [soft assertions](https://assertj.github.io/doc/#assertj-core-soft-assertions), which call `assertAll` automatically so it's not required to call it explicitly in the tests.

**Code Sample demonstrating the issue:**

```
import org.assertj.core.api.SoftAssertions;
import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith({SoftAssertionsExtension.class})
class ShouldIncludeAssertTest {
  @InjectSoftAssertions private SoftAssertions softly;

  @Test
  void falsePositive1() {
    softly.assertThat(false).isTrue();
    softly.assertThat(true).isFalse();
  }

  @Test
  void falsePositive2() {
    SoftAssertions.assertSoftly(
        softly2 -> {
          softly2.assertThat(false).isTrue();
          softly2.assertThat(true).isFalse();
        });
  }
}
```

**Expected outcome:**

No violations reported, but PMD incorrectly reports a violation for both `falsePositive1` and `falsePositive2`. 

**Running PMD through:** Gradle

**Comment by durimkryeziu** [comment]
I have the same usage of soft assertions as in the `falsePositive2()` test; I had to `@SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")` to make it pass for now 😅 

Gradle 8.10
PMD 7.6.0
